src/components/TitlesCarousel/TitlesCarousel.js

```diff
@@ -45,8 +46,14 @@ export const MovieCarousel = async ({ movies, id }) => {
     });
 
     return `
+    ${Button({
+        onClick: () => redirect('/search'),
```

Comment:
You are hardcoding the onClick function here, but actually we need to get this function from arguments, we don't even need to get the function, this component is not very generic, 
Add the "route" argument to the MovieCarousel, and user route from arguments in redirect here